src/main/scala/ai/privado/audit/DataElementDiscovery.scala

```diff
@@ -596,10 +596,11 @@ object DataElementDiscoveryJS {
 
           value.foreach {
             case (member: Member) => {
-              val memberUniqueKey = s"${key.fullName}${key.file.head.name}${member.name}"
-              if (member.name.nonEmpty && !addedMembers.contains(memberUniqueKey)) {
-                addedMembers.add(memberUniqueKey)
-                if (ruleMemberInfo.contains(member.name)) {
+              val memberUniqueKey =
+                s"${key.fullName}${key.file.name.headOption.getOrElse(Constants.EMPTY)}${member.name}"
+              if (member.name.nonEmpty) {
+                if (ruleMemberInfo.contains(member.name) && !addedMembers.contains(memberUniqueKey)) {
```

Comment:
@Saumya25mis The earlier version of handling if condition was correct, this new commit is introducing checking `!addedMembers.contains(memberUniqueKey)` twice

cc - @HiteshMah-Jan